libcsm/hsm/api.py

```diff
@@ -0,0 +1,68 @@
+#
+#  MIT License
+#
+#  (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a
+#  copy of this software and associated documentation files (the "Software"),
+#  to deal in the Software without restriction, including without limitation
+#  the rights to use, copy, modify, merge, publish, distribute, sublicense,
+#  and/or sell copies of the Software, and to permit persons to whom the
+#  Software is furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included
+#  in all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+#  THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+#  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+#  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+#  OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+Submodule for interacting with CSM HSM.
+"""
+
+import http
+from os import getenv
+import requests
+import certifi
+from libcsm import api
+
+ROLE_SUBROLES = ["Management_Master", "Management_Worker", "Management_Storage"]
+
+class API:
+    """
+    Class for providing API to interact with HSM.
+    """
+
+    def __init__(self, api_gateway_address="api-gw-service-nmn.local"):
+
+        self.api_gateway_address = api_gateway_address
+        self.hsm_components_url = f'https://{self.api_gateway_address}/apis/smd/hsm/v2/State/Components'
+        self._auth = api.Auth()
+        self._auth.refresh_token()
+        self._crt_path = getenv("REQUESTS_CA_BUNDLE", certifi.where())
+
+
+    def get_components(self, role_subrole: str):
```

Comment:
Selfishly, please add argument and return annotations. I am going to rebase #37 on this PR and if annotations are present then all the types/values will auto-fill.